src/Language/Go/Term.hs

```diff
@@ -0,0 +1,173 @@
+import qualified Data.Syntax.Literal as Literal
+import qualified Data.Syntax.Statement as Statement
+import qualified Data.Syntax.Type as Type
+import qualified Data.Term as Term
+import Data.Traversable
+import Diffing.Interpreter
+import Language.Go.Syntax as Go.Syntax
+import Language.Go.Type as Go.Type
+import Source.Loc
+import Source.Span
+
+type Syntax =
+  [ Comment.Comment
+  , Declaration.Constructor
+  , Declaration.Function
+  , Declaration.Method
+  , Declaration.MethodSignature
+  , Declaration.Type
+  , Declaration.TypeAlias
+  , Expression.Plus
+  , Expression.Minus
+  , Expression.Times
+  , Expression.DividedBy
+  , Expression.Modulo
+  , Expression.Power
+  , Expression.Negate
+  , Expression.FloorDivision
+  , Expression.BOr
+  , Expression.BAnd
+  , Expression.BXOr
+  , Expression.LShift
+  , Expression.RShift
+  , Expression.UnsignedRShift
+  , Expression.Complement
+  , Expression.Call
+  , Expression.LessThan
+  , Expression.LessThanEqual
+  , Expression.GreaterThan
+  , Expression.GreaterThanEqual
+  , Expression.Equal
+  , Expression.StrictEqual
+  , Expression.Comparison
+  , Expression.Subscript
+  , Expression.Member
+  , Statement.PostDecrement
+  , Statement.PostIncrement
+  , Expression.MemberAccess
+  , Expression.And
+  , Expression.Not
+  , Expression.Or
+  , Expression.XOr
+  , Go.Syntax.Composite
+  , Go.Syntax.DefaultPattern
+  , Go.Syntax.Defer
+  , Go.Syntax.Field
+  , Go.Syntax.Go
+  , Go.Syntax.Label
+  , Go.Syntax.Package
+  , Go.Syntax.Receive
+  , Go.Syntax.ReceiveOperator
+  , Go.Syntax.Rune
+  , Go.Syntax.Select
+  , Go.Syntax.Send
+  , Go.Syntax.Slice
+  , Go.Syntax.TypeAssertion
+  , Go.Syntax.TypeConversion
+  , Go.Syntax.TypeSwitch
+  , Go.Syntax.TypeSwitchGuard
+  , Go.Syntax.Variadic
+  , Go.Type.BidirectionalChannel
+  , Go.Type.ReceiveChannel
+  , Go.Type.SendChannel
+  , Go.Syntax.Import
+  , Go.Syntax.QualifiedImport
+  , Go.Syntax.SideEffectImport
+  , Literal.Array
+  , Literal.Complex
+  , Literal.Float
+  , Literal.Hash
+  , Literal.Integer
+  , Literal.KeyValue
+  , Literal.Pointer
+  , Literal.Reference
+  , Literal.TextElement
+  , Statement.Assignment
+  , Statement.Break
+  , Statement.Continue
+  , Statement.For
+  , Statement.ForEach
+  , Statement.Goto
+  , Statement.If
+  , Statement.Match
+  , Statement.NoOp
+  , Statement.Pattern
+  , Statement.Return
+  , Statement.Statements
+  , Syntax.Context
+  , Syntax.Error
+  , Syntax.Empty
+  , Syntax.Identifier
+  , Type.Annotation
+  , Type.Array
+  , Type.Function
+  , Type.Interface
+  , Type.Map
+  , Type.Parenthesized
+  , Type.Pointer
+  , Type.Slice
+  , []
+  , Literal.String
+  , Literal.EscapeSequence
+  , Literal.Null
+  , Literal.Boolean
+  ]
+
+
+newtype Term ann = Term { getTerm :: Term.TermF (Sum.Sum Syntax) ann (Term ann) }
+  deriving (Eq, Declarations, FreeVariables, Ord, Show, ToJSON)
+
+instance Term.IsTerm Term where
+  type Syntax Term = Sum.Sum Syntax
+  toTermF = coerce
+  fromTermF = coerce
```

Comment:
The new term types are all isomorphic to `Term (Sum Syntax)`, but because they’re monomorphic in the syntax type, we get to save a lot of work: we don’t have to re-check whether or not `Data.Term.Term (Sum …)` is a `Functor` at every call site; instead, `Go.Term` is a functor follows directly from its very existence..